HomeWork/src/main/java/streams/ThreadSafeList.java

```diff
@@ -0,0 +1,61 @@
+package streams;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ThreadSafeList<T> {
+    private final Object lock = new Object();
+    private Object[] array;
+
+    public Object[] getArray() {
+        return array;
+    }
+
+    public void setArray(Object[] array) {
+        this.array = array;
+    }
+
+    public ThreadSafeList() {
+        setArray(new Object[0]);
```

Comment:
1. instead of calling a setter do the assignment directly!
2. Why initialize a list with size 0? Doesn't make any sense if you're planning to add/remove/get some clements across multiple threads.